cpp/thirdparty/CMakeLists.txt

```diff
@@ -51,10 +53,75 @@ endif()
 
 # ## BOOST
 if(MEMILIO_USE_BUNDLED_BOOST)
-    include(BuildBoost)
+    # Ensure that the boost atomic library is used instead of the standard atomic library, where some functionality is only available as of C++20.
+    add_compile_definitions(BOOST_FILESYSTEM_NO_CXX20_ATOMIC_REF)
+    message(STATUS "Downloading Boost library. This may take some time.")
+
+    include(FetchContent)
+    FetchContent_Declare(boost
+    GIT_REPOSITORY https://github.com/boostorg/boost.git
+    GIT_TAG boost-${MEMILIO_BOOST_VERSION})
+    FetchContent_GetProperties(boost)
+
+    if(NOT boost_POPULATED)
+        FetchContent_Populate(boost)
+    endif()
+    
+    if(NOT IS_DIRECTORY "${boost_SOURCE_DIR}/boost")
+        if(MSVC)
+            add_custom_target(boost-bootstrap ALL "bootstrap.bat"
+                COMMAND "b2.exe" headers
+                WORKING_DIRECTORY ${boost_SOURCE_DIR}
+                VERBATIM
+                )
+        else()
+            add_custom_target(boost-bootstrap ALL ./bootstrap.sh
+                COMMAND ./b2 headers 
+                WORKING_DIRECTORY ${boost_SOURCE_DIR}
+                VERBATIM
+                )
+        endif()
+    endif()

```

Comment:
after some thought, this is not ideal.
the existence of the directory is checked at configure time, and if it exists, the target doesn't exist. Thus, if you rerun cmake configuration, the boost-bootstrap target will not exist (because the directory is already there), but the boost target depends on it. Also, the boost directory may be lost for some reason, but bootstrap is not run because the target doesn't exist.

What you really want is for the directory to be checked at build time. CMake can do this automatically. What you need is a custom command that runs bootstrap and a custom target that depends on the boost directory. Here is a simple example https://dev.to/iblancasa/learning-cmake-3-understanding-addcustomcommand-and-addcustomtarget-43gp You should be able to adapt this by setting the DEPENDS argument of the custom target and the OUTPUT argument of the custom command to ${boost_SOURCE_DIR}